pkg/controller/sparkapplication/job.go

```diff
@@ -103,9 +103,11 @@ func (sjm *realSubmissionJobManager) createSubmissionJob(app *v1beta2.SparkAppli
 			BackoffLimit: app.Spec.RestartPolicy.OnSubmissionFailureRetries,
 			Template: corev1.PodTemplateSpec{
 				Spec: corev1.PodSpec{
+					Volumes:          app.Spec.Volumes,
```

Comment:
We probably should limit the `Volumes` to only those appearing on `Driver.VolumeMounts`.

Reply:
And we should also not include volumes for scratch space. See #707.